statina/API/v2/endpoints/batches.py

```diff
@@ -0,0 +1,288 @@
+        return JSONResponse(content="File not found", status_code=404)
+
+    path = Path(file_path)
+    if path.is_dir():
+        file_obj = zip_dir(source_dir=file_path)
+        response = StreamingResponse(file_obj, media_type="application/octet-stream")
+        response.headers["Content-Disposition"] = f"attachment; filename={file_name}"
+        return response
+
+    return FileResponse(
+        str(path.absolute()),
+        media_type="application/octet-stream",
+        filename=file_name or path.name,
+    )
+
+
+@router.get("/batch/{batch_id}/zscore_plot")
+def zscore_plot(
+    batch_id: str,
+    ncv: Literal["13", "18", "21"] = Query(...),
+    current_user: User = Security(get_current_active_user, scopes=["R"]),
+    adapter: StatinaAdapter = Depends(get_nipt_adapter),
+):
+    """Batch view with with Zscore plot"""
+
+    return JSONResponse(
+        content=jsonable_encoder(
+            dict(
+                tris_thresholds=TRISOMI_TRESHOLDS,
+                chromosomes=[ncv],
+                ncv_chrom_data={ncv: get_tris_samples(adapter=adapter, chr=ncv, batch_id=batch_id)},
+                normal_data={ncv: get_tris_control_normal(adapter, ncv)},
+                abnormal_data={ncv: get_tris_control_abnormal(adapter, ncv, 0)},
+            )
+        ),
+    )
+
+
+@router.get("/batch/{batch_id}/fetal_fraction_XY")
+def fetal_fraction_XY(
+    batch_id: str,
+    current_user: User = Security(get_current_active_user, scopes=["R"]),
+    adapter: StatinaAdapter = Depends(get_nipt_adapter),
+):
+    """Batch view with fetal fraction (X against Y) plot"""
+
+    cases = get_fetal_fraction.samples(adapter=adapter, batch_id=batch_id)
+    control: FetalFractionSamples = get_fetal_fraction.samples(
+        batch_id=batch_id, adapter=adapter, control_samples=True
+    )
+    abnormal: FetalFractionControlAbNormal = get_fetal_fraction.control_abnormal(adapter)
+    abnormal_dict = abnormal.dict(
+        exclude_none=True,
+        exclude={
+            "X0": {"status_data_"},
+            "XXX": {"status_data_"},
+            "XXY": {"status_data_"},
+            "XYY": {"status_data_"},
+        },
+    )
+
+    x_max = max(control.FFX + cases.FFX) + 1
+    x_min = min(control.FFX + cases.FFX) - 1
+
+    sex_thresholds = SexChromosomeThresholds(x_min=x_min, x_max=x_max)
+    return JSONResponse(
+        content=jsonable_encoder(
+            dict(
+                sex_thresholds={
+                    "XY_fetal_fraction_y": sex_thresholds.XY_fetal_fraction_y(),
+                    "XX_lower": sex_thresholds.XX_lower(),
+                    "XX_upper": sex_thresholds.XX_upper(),
+                    "XY_upper": sex_thresholds.XY_upper(),
+                    "XY_lower": sex_thresholds.XY_lower(),
+                    "XXY": sex_thresholds.XXY(),
+                },
+                control=control,
+                abnormal=abnormal_dict,
+                cases=cases,
+                max_x=x_max,
```

Comment:
I guess you removed the batch because it can be called from another endpoint in the same view?

Reply:
Yes you can get batch with another call at the same batch id